File: dotnet/controller-api/OrderStore.cs
using System.Collections.Concurrent;

namespace controller_api;

public class OrderStore
{
    private readonly ConcurrentDictionary<string, OrderResponse> _store = new();
    private int _counter;

    public string NextOrderId() => Interlocked.Increment(ref _counter).ToString();

    public static string Key(string userId, string orderId) => $"{userId}:{orderId}";

    public void Set(string key, OrderResponse order) => _store[key] = order;

    public bool TryGet(string key, out OrderResponse? order) => _store.TryGetValue(key, out order);

    public bool Contains(string key) => _store.ContainsKey(key);

    public bool TryRemove(string key) => _store.TryRemove(key, out _);

    public List<OrderResponse> GetByUser(string userId)
    {
        var prefix = $"{userId}:";
        var results = new List<OrderResponse>();
        foreach (var kvp in _store)
        {
            if (kvp.Key.StartsWith(prefix))
                results.Add(kvp.Value);
        }
        return results;
    }
}

public class ProfileStore
{
    private readonly ConcurrentDictionary<string, UserProfile> _store = new();

    public void Set(string userId, UserProfile profile) => _store[userId] = profile;

    public bool TryGet(string userId, out UserProfile? profile) => _store.TryGetValue(userId, out profile);
}

File: java/quarkus/src/main/java/benchmark/model/OrderStore.java
package benchmark.model;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.atomic.AtomicLong;

public class OrderStore {
    private static final OrderStore INSTANCE = new OrderStore();
    private final ConcurrentHashMap<String, OrderResponse> store = new ConcurrentHashMap<>();
    private final ConcurrentHashMap<String, UserProfile> profileStore = new ConcurrentHashMap<>();
    private final AtomicLong counter = new AtomicLong(0);

    private OrderStore() {}

    public static OrderStore getInstance() {
        return INSTANCE;
    }

    public long nextId() {
        return counter.incrementAndGet();
    }

    public void put(String userId, String orderId, OrderResponse order) {
        store.put(userId + ":" + orderId, order);
    }

    public OrderResponse get(String userId, String orderId) {
        return store.get(userId + ":" + orderId);
    }

    public OrderResponse remove(String userId, String orderId) {
        return store.remove(userId + ":" + orderId);
    }

    public List<OrderResponse> listByUser(String userId) {
        String prefix = userId + ":";
        List<OrderResponse> results = new ArrayList<>();
        store.forEach((key, value) -> {
            if (key.startsWith(prefix)) {
                results.add(value);
            }
        });
        return results;
    }

    public void putProfile(String userId, UserProfile profile) {
        profileStore.put(userId, profile);
    }

    public UserProfile getProfile(String userId) {
        return profileStore.get(userId);
    }
}
